io.openems.edge.core/src/io/openems/edge/core/meta/MetaImpl.java

```diff
@@ -59,4 +62,14 @@ protected void deactivate() {
 		super.deactivate();
 	}
 
+	private void applyConfig(Config config) {
+		this._setCurrency(this.getCurrency(config));
+	}
+
+	private Currency getCurrency(Config config) {
```

Comment:
The way you implemented it, it should be static, because it has no reference to `this`. 

I would instead prefer to move this function to the `CurrencyConfig` class. Add a `toCurrency()` method that translates the 'current' CurrencyConfig to a `Currency`.

This way you can simply use it above using: `this._setCurrency(config.currency().toCurrency())`;